code/game/objects/structures/divine.dm

```diff
@@ -6,19 +23,47 @@
 	anchored = TRUE
 	density = FALSE
 	can_buckle = 1
+	max_integrity = 30000
+	var/divine_chance = 5
 
-/obj/structure/sacrificealtar/on_attack_hand(mob/living/user, act_intent = user.a_intent, unarmed_attack_flags)
+/obj/structure/sacrificealtar/attackby(obj/item/I, mob/user)
 	. = ..()
+	var/divine = prob(divine_chance)
 	if(.)
 		return
+	if(istype(I, /obj/item/gun))
+		to_chat(user, "<span class='notice'>You attempt to sacrifice [I] by invoking the sacrificial ritual, it melts and breaks into pieces.</span>")
+		qdel(I)
+		message_admins("[ADMIN_LOOKUPFLW(user)] has sacrificed [key_name_admin(I)] on the sacrificial altar at [AREACOORD(src)].")
+		switch(divine)
+			if(FALSE)
+				var/loot_item = pick(GLOB.gunparts_rates)
+				new loot_item (get_turf(user))
+				return
+			if(TRUE)
+				var/pick_adcomp = pickweight(GLOB.loot_craft_advanced)
+				new pick_adcomp(get_turf(user))
+				return
 	if(!has_buckled_mobs())
 		return
 	var/mob/living/L = locate() in buckled_mobs
 	if(!L)
 		return
-	to_chat(user, "<span class='notice'>You attempt to sacrifice [L] by invoking the sacrificial ritual.</span>")
-	L.gib()
-	message_admins("[ADMIN_LOOKUPFLW(user)] has sacrificed [key_name_admin(L)] on the sacrificial altar at [AREACOORD(src)].")
+	if(istype(I, /obj/item/clothing/accessory/talisman))
+		to_chat(user, "<span class='notice'>You attempt to sacrifice [L] by invoking the sacrificial ritual, during the sacrifice, you removed something from the stomach of the victim.</span>")
```

Comment:
I'd prefer if you had to kill the mob first. Or just have that be a requirement for non-simple mobs. 

I also dont like the idea of straight up gibbing player mobs, I'd prefer something like marking them as sacrificed so they cant be sacrificed again, or destroying their brain to keep them dead for good.

Reply:
Good idea, but I don't know how to do it. If someone tells me the code to remove the brain of the victim, I will gladly write it.

but I added a check that the victim must be dead.

Reply:
Should be okay for now, I see it mostly just for livestock, with the very occasional player being sacrificed for RP reasons.